crystal_diffusion/models/mtp.py

```diff
@@ -0,0 +1,315 @@
+                 ) -> Tuple[pd.DataFrame, pd.DataFrame]:
+        """Evaluate energies, forces, stresses and MaxVol gamma factor of structures with trained MTP.
+
+        Args:
+            test_structures: evaluation set of pymatgen Structure Objects.
+            test_energies: list of total energies of each structure to evaluation in test_structures list.
+            test_forces: list of calculated (m, 3) forces of each evaluation structure with m atoms in structures list.
+                m can be varied with each single structure case.
+            test_stresses (optional): list of calculated (6, ) virial stresses of each evaluation structure in
+                test_structures list. If None, do not evaluate on stresses. Default to None.
+
+        Returns:
+            dataframe with ground truth energies, forces
+            dataframe with predicted energies, forces, MaxVol gamma (nbh grades)
+        """
+        original_file = "original.cfgs"
+        predict_file = "predict.cfgs"
+        test_structures, test_forces, test_stresses = check_structures_forces_stresses(
+            test_structures, test_forces, test_stresses
+        )
+        predict_pool = pool_from(test_structures, test_energies, test_forces, test_stresses)
+
+        with ScratchDir("."):  # mlip needs a tmp_work_dir - we will manually copy relevant outputs elsewhere
+            # write the structures to evaluate in a mlp compatible format
+            original_file = self.write_cfg(original_file, cfg_pool=predict_pool)
+            df_orig = self.read_cfgs(original_file, nbh_grade=False)  # read original values as a DataFrame
+
+            # calculate_grade is the method to get the forces, energy & maxvol values
+            cmd = [self.mlp_command, "calculate_grade", self.fitted_mtp, original_file, predict_file]
+            predict_file += '.0'  # added by mlp...
+            with subprocess.Popen(cmd, stdout=subprocess.PIPE) as p:  # run mlp
+                stdout = p.communicate()[0]
+                rc = p.returncode
+            if rc != 0:
+                error_msg = f"mlp exited with return code {rc}"
+                msg = stdout.decode("utf-8").split("\n")[:-1]
+                try:
+                    error_line = next(i for i, m in enumerate(msg) if m.startswith("ERROR"))
+                    error_msg += ", ".join(msg[error_line:])
+                except Exception:
+                    error_msg += msg[-1]
+                raise RuntimeError(error_msg)
+
+            df_predict = self.read_cfgs(predict_file, nbh_grade=True)
+        return df_orig, df_predict
+
+    def read_cfgs(self, filename: str, nbh_grade: bool = False) -> pd.DataFrame:
+        """Read mlp output when MaxVol gamma factor is present.
+
+        Args:
+            filename: name of mlp output file to be parsed.
+            nbh_grade (optional): if True, add the nbh_grades in the resulting dataframe. Defaults to False.
```

Comment:
what does "nbh_grade" mean? Is "nbh" an acronym? A quick explanation here would be useful.

Reply:
nbh stands for neighborhood. mlip does either structure or neighborhood to compute the gamma factor. Let's stick to nbh (default) for now. We could revisit this later. 
Added information in the arg description.